### ISSUE TITLE
Popover not working when created together with target

### ISSUE THREAD

**Issue by luiztauffer** [facultyai/dash-bootstrap-components#716]
Before opening:

- [Search for duplicate or closed issues](https://github.com/facultyai/dash-bootstrap-components/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the [contributing guidelines](https://github.com/facultyai/dash-bootstrap-components/blob/main/.github/CONTRIBUTING.md)

Please fill out the below information as much as possible.

- dash version: `2.0.0`
- dash-bootstrap-components version: `1.0.0b3`
- components affected by bug: Popover

### What is happening?

Popover doesn't seem to work when created with the Target component at the same time (same callback call).

### What should be happening?

Popover should appear when hovering (or other action) on target.

### Code

Previously, with `dash=1.21.0` and `dash_boostrap_components=0.13.0`, a code like this worked just fine: (see GIF below)
```python
import dash_bootstrap_components as dbc
from dash_bootstrap_components._components.Label import Label
import dash_html_components as html
import dash
from dash.dependencies import Output, Input, State

app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])

popover = html.Div(
    [   
        dbc.Label(f'Dash: {dash.__version__}'),
        html.Br(),
        dbc.Label(f'DBC: {dbc.__version__}'),
        html.Br(),
        dbc.Button(
            "Create",
            id="create",
            color="success",
            className="mr-4",
            n_clicks=0,
        ),
        html.Div(children=[], id='div-targets'),
        html.Div(children=[], id='div-popovers'),
    ]
)

app.layout = dbc.Container(popover)

@app.callback(
    [Output("div-targets", "children"), Output("div-popovers", "children")],
    [Input("create", "n_clicks")],
    [State('div-targets', 'children'), State('div-popovers', 'children')]
)
def create_popover(n, targets, popovers):
    if n:
        targets.append(dbc.Button("Target", id=f"target-{n}", color="danger", n_clicks=0))

        popovers.append(
            dbc.Popover(
                [
                    dbc.PopoverHeader(f"Popover header: {n}"),
                    dbc.PopoverBody(f"Number {n}"),
                ],
                id=f"popover-{n}",
                target=f"target-{n}",
                is_open=False,
                trigger="hover"
            )
        )
        return targets, popovers
    return dash.no_update, dash.no_update

if __name__ == "__main__":
    app.run_server(debug=True)
```
![working](https://user-images.githubusercontent.com/24541631/134933822-1b824a38-c764-4133-84d3-7a876f48d9cd.gif)

For the same code, but with `dash=2.0.0` and `dash_boostrap_components=1.0.0b3`, it doesn't work anymore:

![not_working_1](https://user-images.githubusercontent.com/24541631/134934581-ab7f5679-c79a-4283-b2f6-72e470384054.gif)

Now, I tested with this small change, to use as target one element already created, and it works...
```python
popovers.append(
    dbc.Popover(
        [
            dbc.PopoverHeader(f"Popover header: {n}"),
            dbc.PopoverBody(f"Number {n}"),
        ],
        id=f"popover-{n}",
        target=f"target-{n-1}",
        is_open=False,
        trigger="hover"
    )
)
```

![not_working_2](https://user-images.githubusercontent.com/24541631/134935161-a0453f69-414b-4e41-a0a5-15bb165247f9.gif)

### Error messages

No error messages appear.

**Comment by tcbegley** [comment]
Thanks for raising this @luiztauffer 

I think the popover only looks for the target once when the component is created. Perhaps this can be fixed by making the Popover look for the target repeatedly at increasing intervals until it finds something. Will look into it!

**Comment by luiztauffer** [comment]
thanks for looking into it @tcbegley 
FYI, the exact same problem happens with `dbc.Tooltip`, I just tested here

**Comment by tcbegley** [comment]
Thanks! Yes that's expected as the same component controls the rendering, they differ only in what they render, but helpful to know it is the case.

**Comment by luiztauffer** [comment]
do you think it could be something on Dash side? Since it was working for `dash<2.0`...

**Comment by tcbegley** [comment]
No, the implementation in dash-bootstrap-components has changed as we migrated to react-bootstrap from reactstrap (the latter is not very actively maintained anymore and is being slow to support Bootstrap 5, the former is much more actively maintained and feature-complete + results in smaller bundle sizes).

Actually the new implementation we have much more control over, and has enabled us to properly support dictionary ids like those that are used in pattern-matching callbacks, so I'm cautiously optimistic that we will be able to adjust the implementation to support this before the full release.

**Comment by tcbegley** [comment]
This should be resolved in `1.0.0rc1`. Let me know if you still have problems.

```
pip install -U --pre dash-bootstrap-components
```

![ezgif-4-cea853324fa2](https://user-images.githubusercontent.com/15220906/136653548-6e37869d-0058-46e2-bcd3-5c3582987e86.gif)
